internal/service/vmservice/vm.go

```diff
@@ -371,6 +371,17 @@ func createVM(ctx context.Context, scope *scope.MachineScope) (proxmox.VMCloneRe
 	}
 
 	templateID := scope.ProxmoxMachine.GetTemplateID()
+	if templateID == -1 {
+		var err error
+
+		templateSelectorTags := scope.ProxmoxMachine.GetTemplateSelectorTags()
+		options.Node, templateID, err = scope.InfraCluster.ProxmoxClient.FindVMTemplateByTags(ctx, templateSelectorTags)
+
+		if err != nil {
+			scope.SetFailureMessage(err)
```

Comment:
You don't need this Failure unless you know 100% that the errors is caused by template not found!

Reply:
and please write better message with an error reason.

Reply:
missing `FailureReason`